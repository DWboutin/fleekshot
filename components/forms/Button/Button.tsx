import React, { ReactNode } from "react";
import styled from "styled-components";

import { ThemeContainer } from "../../../styles/styles";
import { toREM } from "../../../styles/typography";

interface ContainerProps {}

const Container = styled.button<ContainerProps>`
  font-size: ${toREM(14)};
  border: 0;
  padding: 5px 9px;
  font-weight: 600;

  ${({ theme }: ThemeContainer) => `
    background-color: ${theme.forms.button.bg};
    border-radius: ${theme.forms.button.borderRadius};
    color: ${theme.forms.button.color};
  `}

  &:hover {
    background-color: ${({ theme }: ThemeContainer) =>
      theme.forms.button.hoverBg};
  }
  &:active {
    background-color: ${({ theme }: ThemeContainer) =>
      theme.forms.button.activeBg};
  }
`;

export type ButtonTypes = "submit" | "button";

export interface Props {
  children: ReactNode;
  type?: ButtonTypes;
}

const Button: React.VoidFunctionComponent<Props> = ({
  children,
  type = "button",
}) => {
  return <Container type={type}>{children}</Container>;
};

export default Button;
