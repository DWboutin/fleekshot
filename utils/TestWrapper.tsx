import React, { ReactNode } from "react";
import styled, { ThemeProvider } from "styled-components";
import IntlManager from "../components/IntlManager";
import GlobalStyle from "../styles/globalStyle";

import { theme } from "../styles/styles";

interface Props {
  children: ReactNode;
}

const TestWrapper: React.FunctionComponent<Props> = ({ children }) => (
  <ThemeProvider theme={theme}>
    <GlobalStyle />
    <IntlManager>{children}</IntlManager>
  </ThemeProvider>
);

export default TestWrapper;
