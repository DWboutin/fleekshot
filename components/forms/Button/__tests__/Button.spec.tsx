import {
  fireEvent,
  render,
  RenderResult,
  screen,
} from "@testing-library/react";
import TestWrapper from "../../../../utils/TestWrapper";

import Button, { Props } from "../Button";

describe("<Button />", () => {
  let renderResult: RenderResult;

  const renderButton = ({ children }: Props) => {
    renderResult = render(
      <TestWrapper>
        <Button>{children}</Button>
      </TestWrapper>
    );
  };

  describe("default rendering", () => {
    const CHILDREN = "children";

    beforeEach(() => {
      renderButton({
        children: CHILDREN,
      });
    });

    it("should match snapshot", () => {
      expect(renderResult.container).toMatchSnapshot();
    });

    it(`should contain the text "${CHILDREN}"`, () => {
      expect(screen.getByText(CHILDREN)).toBeInTheDocument();
    });
  });
});
