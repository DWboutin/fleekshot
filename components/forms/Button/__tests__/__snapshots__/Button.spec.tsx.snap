// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> default rendering should match snapshot 1`] = `
<div>
  <button
    class="sc-bdvvtL cApkZc"
    type="button"
  >
    <span>
      children
    </span>
  </button>
</div>
`;
